apps/meteor/client/views/admin/apps/AppDetailsPage.tsx

```diff
@@ -18,59 +21,122 @@ const AppDetailsPage: FC<{ id: string }> = function AppDetailsPage({ id }) {
 
 	const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
 	const [isSaving, setIsSaving] = useState(false);
-	const settingsRef = useRef<Record<string, ISetting['value']>>({});
+	const [selectedTab, setSelectedTab] = useState({
+		isSelectedDetails: true,
+		isSelectedLogs: false,
+		isSelectedSettings: false,
+	});
 
+	const settingsRef = useRef<Record<string, ISetting['value']>>({});
 	const data = useAppInfo(id);
 
 	const [currentRouteName] = useCurrentRoute();
 	if (!currentRouteName) {
 		throw new Error('No current route name');
 	}
-	const router = useRoute(currentRouteName);
-	const handleReturn = (): void => router.push({});
 
-	const { settings, apis } = { settings: {}, apis: [], ...data };
+	const router = useRoute(currentRouteName);
+	const handleReturn = useMutableCallback((): void => router.push({}));
 
-	const showSettings = Object.values(settings).length;
-	const showApis = apis.length;
+	const { installed, settings, apis } = data || {};
+	const showApis = apis?.length;
 
 	const saveAppSettings = useCallback(async () => {
 		const { current } = settingsRef;
 		setIsSaving(true);
 		try {
 			await Apps.setAppSettings(
 				id,
-				(Object.values(settings) as ISetting[]).map((value) => ({ ...value, value: current?.[value.id] })) as unknown as ISettingsPayload,
+				(Object.values(settings || {}) as ISetting[]).map((value) => ({
+					...value,
+					value: current?.[value.id],
+				})) as unknown as ISettingsPayload,
 			);
 		} catch (e) {
 			handleAPIError(e);
 		}
 		setIsSaving(false);
 	}, [id, settings]);
 
+	const isClickedTabDetails = useCallback(
+		(e: React.MouseEvent<HTMLElement, MouseEvent>): boolean => e.currentTarget.innerText === t('Details'),
+		[t],
+	);
+
+	const isClickedTabLogs = useCallback(
+		(e: React.MouseEvent<HTMLElement, MouseEvent>): boolean => e.currentTarget.innerText === t('Logs'),
+		[t],
+	);
+
+	const isClickedTabSettings = useCallback(
+		(e: React.MouseEvent<HTMLElement, MouseEvent>): boolean => e.currentTarget.innerText === t('Settings'),
+		[t],
+	);
+
+	const selectTab: MouseEventHandler<HTMLElement> = useCallback(
+		(e) => {
+			setSelectedTab({
+				isSelectedDetails: isClickedTabDetails(e),
+				isSelectedLogs: isClickedTabLogs(e),
+				isSelectedSettings: isClickedTabSettings(e),
+			});
+		},
+		[isClickedTabDetails, isClickedTabLogs, isClickedTabSettings],
+	);
+
```

Comment:
just dont do this, we use react for states , dont manipulate directly html and inner values.

 look at your `selectedTab` the current type is 

```
{
	isSelectedDetails: boolean,
	isSelectedLogs: boolean,
	isSelectedSettings: booelan,
}
```
looks a little messy to perform/check what you are doing

why not `selectedTab: 'details' | 'logs' | 'settings'` ?

and then 
```
const isClickedTabLogs = useCallback(() => selectTab('log'));
```

btw why not do this based on the url?